src/test/java/io/supertokens/test/userMetadata/UserMetadataTest.java

```diff
@@ -0,0 +1,173 @@
+            return;
+        }
+
+        String userId = "userId";
+        JsonObject metadata = UserMetadata.getUserMetadata(process.getProcess(), userId);
+
+        assertNotNull(metadata);
+        assertEquals(metadata.entrySet().size(), 0);
+
+        process.kill();
+        assertNotNull(process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.STOPPED));
+    }
+
+    /**
+     * set the metadata for a user who doesn't have one
+     *
+     * @throws Exception
+     */
+    @Test
+    public void createMetadata() throws Exception {
+        String[] args = { "../" };
+
+        TestingProcessManager.TestingProcess process = TestingProcessManager.start(args);
+        assertNotNull(process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.STARTED));
+
+        if (StorageLayer.getStorage(process.getProcess()).getType() != STORAGE_TYPE.SQL) {
+            return;
+        }
+
+        String userId = "userId";
+        JsonObject update = new JsonObject();
+        update.addProperty("test", "123");
+        update.add("testNull", JsonNull.INSTANCE);
+        JsonObject updateResult = UserMetadata.updateUserMetadata(process.getProcess(), userId, update);
+
+        JsonObject newMetadata = UserMetadata.getUserMetadata(process.getProcess(), userId);
+
+        assertEquals(updateResult, newMetadata);
+        assertEquals(1, updateResult.entrySet().size());
+        assertEquals("123", updateResult.get("test").getAsString());
+        assert (!updateResult.has("testNull"));
+
+        process.kill();
+        assertNotNull(process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.STOPPED));
+    }
+
+    /**
+     * updates the metadata of a user and merges the result
+     *
+     * @throws Exception
+     */
+    @Test
+    public void updateMetadata() throws Exception {
+        String[] args = { "../" };
+
+        TestingProcessManager.TestingProcess process = TestingProcessManager.start(args);
+        assertNotNull(process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.STARTED));
+
+        if (StorageLayer.getStorage(process.getProcess()).getType() != STORAGE_TYPE.SQL) {
+            return;
+        }
+
+        String userId = "userId";
+
+        JsonObject originalMetadata = new JsonObject();
+        JsonObject subObject = new JsonObject();
+        subObject.addProperty("subsub", "123");
+        originalMetadata.add("testUpdate", subObject);
+        originalMetadata.addProperty("unmodified", "123");
+        originalMetadata.addProperty("cleared", 123);
+
+        // First we create the original
+        UserMetadata.updateUserMetadata(process.getProcess(), userId, originalMetadata);
+
+        JsonObject update = new JsonObject();
+        JsonObject updateSubObject = new JsonObject();
+        updateSubObject.addProperty("subsubupdate", "subnew!");
+        update.add("testUpdate", updateSubObject);
+        update.addProperty("testNew", "new!");
+        update.add("cleared", JsonNull.INSTANCE);
```

Comment:
You shold also try by just doing `update.add("cleared", null);`?

Reply:
That does the same thing in effect: the `add` function translates the null into a JsonNull.